src/Connector.SqlServer/Connector/ISqlClient.cs

```diff
@@ -5,8 +5,12 @@
 
 namespace CluedIn.Connector.SqlServer.Connector
 {
+    public record ConnectionConfigurationError(string ErrorMessage);
```

Comment:
nit: Move to separate file

Reply:
We prefer to keep here, to not litter project with excessive files. We commonly keep records of this kind in same file as the interface that uses them.